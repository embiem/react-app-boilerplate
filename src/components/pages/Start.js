import React, { Component } from "react";
import { connect } from "react-redux";

export class Start extends Component {
  render() {
    return (
      <div className="start">
        <h1>Start</h1>
        <p>{this.props.count}</p>
        <button onClick={this.props.increment}>Increment</button>
      </div>
    );
  }
}

export default connect(
  state => ({
    count: state.count
  }),
  dispatch => ({
    increment: () => dispatch.count.incrementAsync(2)
  })
)(Start);
