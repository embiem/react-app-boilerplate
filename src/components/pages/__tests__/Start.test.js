import React from "react";
import renderer from "react-test-renderer";
import { Start } from "../Start";

test("Start renders without crashing", () => {
  const component = renderer.create(<Start />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
