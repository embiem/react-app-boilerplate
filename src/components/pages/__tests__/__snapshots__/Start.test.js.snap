// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start renders without crashing 1`] = `
<div
  className="start"
>
  <h1>
    Start
  </h1>
  <p />
  <button
    onClick={undefined}
  >
    Increment
  </button>
</div>
`;
